### DISCUSSION TITLE
Generic transports

### DISCUSSION THREAD

**Discussion by roshanrags** [tokio-rs/axum#2674] [Ideas]
It seems like TcpListener/TcpStream are pretty deeply embedded in the current high level APIs. I know the serve-with-hyper example exists, but it seems to essentially be copying (axum internals?) `into_future` just to get a few customization points. It is also something that would need periodic updating as `into_future` gets more and more sophisticated.

Are generic transports something you would be interested in? The goal would basically be so users can supply their own listeners and/or streams to high level APIs and have axum run transparently on top of them.

**Comment by jplatte** [comment]
Yes, it is planned for 0.8: https://github.com/tokio-rs/axum/pull/2479

**Reply by roshanrags** [reply]
Perfect, thanks!
